implementations/rs/ockam/src/profile/profile_manager.rs

```diff
@@ -0,0 +1,58 @@
+use crate::profile::profile::Profile;
+use crate::profile::profile_entry::ProfileEntry;
+use crate::profile::{Identity, ProfileVault};
+use ockam_common::error::OckamResult;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+
+pub struct ProfileManager {}
+
+impl ProfileManager {
+    pub fn new() -> Self {
+        ProfileManager {}
+    }
+
+    pub fn create_profile(
+        &self,
+        identity: Option<Identity>,
+        additional_data: HashMap<String, String>,
+        vault: Arc<Mutex<dyn ProfileVault>>,
+    ) -> OckamResult<Profile> {
+        let entry = ProfileEntry::new(identity.clone(), false, None, vault.clone())?;
+        let mut entries: Vec<ProfileEntry> = Vec::with_capacity(1);
+        entries.push(entry);
+        let profile = Profile::new(identity, entries, additional_data, vault);
+
+        Ok(profile)
+    }
+
+    pub fn rotate_profile(&self, profile: &mut Profile) -> OckamResult<()> {
+        profile.rotate()
+    }
+
+    pub fn revoke_profile(&self, profile: &mut Profile) -> OckamResult<()> {
+        profile.revoke()
```

Comment:
Should we also perform delete?

Reply:
Yes. Delete implies all data about the profile.